app/javascript/react/components/EditActivitiesContainer.js

```diff
@@ -0,0 +1,114 @@
+import React, { useState, useEffect } from 'react'
+import { Redirect } from 'react-router-dom'
+
+import ActivityInputTile from './ActivityInputTile'
+
+const EditActivitiesContainer = props => {
+  const [shouldRedirect, setShouldRedirect] = useState(false)
+  const [activities, setActivities] = useState([])
+  const [shouldRender, setShouldRender] = useState(false)
+  const [activityInfo, setActivityInfo] = useState({})
+
+  useEffect(() => {
+    fetch(`/api/v1/workouts/${props.match.params.id}/activities`)
+    .then((response) => {
+      if (response.ok) {
+        return response
+      } else {
+        let errorMessage = `${response.status} (${response.statusText})`,
+          error = new Error(errorMessage)
+        throw(error)
+      }
+    })
+    .then(response => response.json())
+    .then(body => {
+      setActivities(body.activities)
+      let initialObject = {}
+      body.activities.forEach(activity => {
+        initialObject = {
+          ...initialObject,
+          [activity.id]: {reps: "", sets: "", weight: "", unit: ""}
+        }
+      })
```

Comment:
ideally, this complex object is already created on the backend so that we dont need to set it up here in the .then